src/components/NewEventForm/NewEventForm.tsx

```diff
@@ -0,0 +1,218 @@
+      .string()
+      .max(200, "Location text is too long"),
+    imageUrl: Yup
+      .string()
+      .url("URL not valid"),
+    eventStatus: Yup
+      .string()
+      .oneOf([EVENT_STATUS_OPTIONS.Draft, EVENT_STATUS_OPTIONS.Published])
+      .required('Event status is required')
+  });
+
+  return (
+    <div>
+      <Formik
+        initialValues={InitialValues}
+        validationSchema = {EventFormSchema}
+        onSubmit={async (values, { setSubmitting }) => {
+          await new Promise(r => {
+            setSubmitting(true);
+      
+            try {
+              onSubmit ({
+                id : rest.id,
+                name : values.name,
+                description : values.description,
+                startTime : parseDate(eventDate, values.startTime),
+                endTime : parseDate(eventDate, values.endTime),
+                imageUrl : values.imageUrl,
+                imageAlt : values.imageAlt,
+                location : values.location,
+                eventType : values.eventType,
+                eventStatus : values.eventStatus,
+                teamsUrl : values.teamsUrl,    
+              });    
+            } catch (error) {}
+          });
+          setSubmitting(false);
+        }}
+      >
+        <EventFormWrapper>
+          <h1>{rest.headerTitle}</h1>
+          <Form>
+            <FormikInputField
+              label = "Event name *"
+              name = "name"
+              type = "text"
+            />
+            <FormikSelectorField
+              label="Event type *"
+              name = "eventType"
+              options={EVENT_OPTIONS}
+            />
+            <FormikInputField
+              label = "Date *"
+              name = "date"
+              type = "date"
+            />
+            <FormikInputField
+              label = "Start time *"
+              name = "startTime"
+              type = "time"
+            />
+            <FormikInputField
+              label = "End time *"
+              name = "endTime"
+              type = "time"
+            />
+            <FormikTextAreaField
+              label = "Description *"
+              name = "description"
+              type = "text"
+            />
+            <FormikInputField
+              label = "Location"
+              name = "location"
+              type = "text"
+            />
+            <FormikInputField
+              label = "Teams URL"
+              name = "teamsUrl"
+              type = "text"
+              placeholder = "https://teams.microsoft.com/..."
+            />
+            <FormikInputField
+              label = "Image Link"
+              name = "imageUrl"
+              type = "text"
+            />
+            <FormikInputField
+              label = "Image Text"
+              name = "imageAlt"
+              type = "text"
+            />
+            <div style={{justifyContent:'center', display:'flex', alignItems: 'center', marginTop:'1rem'}}> 
```

Comment:
This styling should be extraced into a styled component. ref. the comment from @sebastto